gooddata-sdk/gooddata_sdk/catalog/workspace/model/workspace.py

```diff
@@ -6,24 +6,44 @@
 from gooddata_metadata_client.model.json_api_workspace_in import JsonApiWorkspaceIn
 from gooddata_metadata_client.model.json_api_workspace_in_attributes import JsonApiWorkspaceInAttributes
 from gooddata_metadata_client.model.json_api_workspace_in_document import JsonApiWorkspaceInDocument
+from gooddata_metadata_client.model.json_api_workspace_in_relationships import JsonApiWorkspaceInRelationships
+from gooddata_metadata_client.model.json_api_workspace_in_relationships_parent import (
+    JsonApiWorkspaceInRelationshipsParent,
+)
+from gooddata_metadata_client.model.json_api_workspace_to_one_linkage import JsonApiWorkspaceToOneLinkage
 from gooddata_sdk.catalog.entity import CatalogNameEntity
 
 
 class CatalogWorkspace(CatalogNameEntity):
+    def __init__(self, workspace_id: str, name: str, parent_id: str = None):
+        super(CatalogWorkspace, self).__init__(workspace_id, name)
+        self.parent_id = parent_id
+
     @classmethod
     def from_api(cls, entity: dict[str, Any]) -> CatalogWorkspace:
         ea = entity["attributes"]
+        er = entity.get("relationships")
         return cls(
-            id=entity["id"],
-            name=ea["name"],
+            workspace_id=entity["id"], name=ea["name"], parent_id=er.get("parent").get("data").get("id") if er else None
```

Comment:
Schema of response is designed in a way that `parent` key of `relationships` is optional. `er.get("parent").get...` could fail with `AttributeError: 'NoneType' object has no attribute 'get'`.

Reply:
I discussed this with @jaceksan and when workspace has relationships field then it definitely has the following structure (parent -> data –> id).

Reply:
Schema is not in our favor. It seems we are 100% sure it is implemented in GD.CN the way you mentioned. Consider this comment covered.